interoperator/pkg/controller/sfserviceinstance/sfserviceinstance_controller.go

```diff
@@ -333,3 +336,52 @@ func removeString(slice []string, s string) (result []string) {
 	}
 	return
 }
+
+func (r *ReconcileServiceInstance) handleError(object *osbv1alpha1.SFServiceInstance, result reconcile.Result, inputErr error) (reconcile.Result, error) {
+	annotations := object.GetAnnotations()
+	var count int64
+	id := object.GetName()
+
+	if inputErr == nil {
+		count = 0
+	} else {
+		countString, ok := annotations[errorCountKey]
+		if !ok {
+			count = 0
+		} else {
+			i, err := strconv.ParseInt(countString, 10, 64)
+			if err != nil {
+				count = 0
+			} else {
+				count = i
+			}
+		}
+		count++
+	}
+
+	if count > errorThreshold {
+		log.Printf("Error threshold reached for %s. Ignoring %v\n", id, inputErr)
```

Comment:
Can we set the status of `SFServiceInstance` and `SFServiceBinding` as `failed` after 10 retries?

Reply:
Updated the PR

Reply:
@vivekzhere Can you please update other properties in crds as well like `error` and `description` else it will be too difficult to figure out what the heck happened